File: src/main/java/com/ggardet/hospital/spi/mapper/TreatmentMapper.java
package com.ggardet.hospital.spi.mapper;

import com.ggardet.hospital.internal.entity.Treatment;
import com.ggardet.hospital.spi.dto.TreatmentDTO;
import org.mapstruct.Mapper;
import org.mapstruct.MappingConstants;

import java.util.List;

@Mapper(componentModel = MappingConstants.ComponentModel.SPRING)
public interface TreatmentMapper {
    TreatmentDTO toDto(final Treatment treatment);

    Treatment toEntity(final TreatmentDTO treatmentDTO);

    List<TreatmentDTO> toDtos(final List<Treatment> treatments);
}

File: src/main/kotlin/com/ggardet/hospital/spi/mapper/TreatmentMapper.kt
package com.ggardet.hospital.spi.mapper

import com.ggardet.hospital.internal.entity.Treatment
import com.ggardet.hospital.spi.dto.TreatmentDTO
import org.springframework.stereotype.Component

@Component
class TreatmentMapper {
    
    fun toDto(treatment: Treatment): TreatmentDTO {
        return TreatmentDTO(
            id = treatment.id,
            name = treatment.name,
            description = treatment.description,
            durationMinutes = treatment.durationMinutes,
            protocol = treatment.protocol
        )
    }

    fun toEntity(treatmentDTO: TreatmentDTO): Treatment {
        return Treatment(
            id = treatmentDTO.id,
            name = treatmentDTO.name,
            description = treatmentDTO.description,
            durationMinutes = treatmentDTO.durationMinutes ?: 0,
            protocol = treatmentDTO.protocol
        )
    }

    fun toDtos(treatments: List<Treatment>): List<TreatmentDTO> {
        return treatments.map(::toDto)
    }
}